lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -873,113 +871,43 @@ updateTx (Cardano.Tx body existingKeyWits) extraContent = do
 -- to treat Alonzo and Shelley differently.
 modifyShelleyTxBody
     :: TxUpdate
-    -> ShelleyBasedEra era
+    -> RecentEra era
     -> Ledger.TxBody (Cardano.ShelleyLedgerEra era)
     -> Ledger.TxBody (Cardano.ShelleyLedgerEra era)
 modifyShelleyTxBody txUpdate era ledgerBody = case era of
-    ShelleyBasedEraConway -> ledgerBody
+    RecentEraConway -> ledgerBody
         { Conway.outputs = Conway.outputs ledgerBody
             <> StrictSeq.fromList
-                ( Ledger.mkSized
-                . Cardano.toShelleyTxOut era
-                . Cardano.toCtxUTxOTxOut
-                . toCardanoTxOut era <$> extraOutputs
-                )
+                (Ledger.mkSized . toConwayTxOut <$> extraOutputs)
         , Conway.inputs = Conway.inputs ledgerBody
             <> Set.fromList (Cardano.toShelleyTxIn <$> extraInputs')
         , Conway.collateral = Conway.collateral ledgerBody
             <> Set.fromList (Cardano.toShelleyTxIn <$> extraCollateral')
         , Conway.txfee =
             modifyFee $ Conway.txfee ledgerBody
         }
-    ShelleyBasedEraBabbage -> ledgerBody
+    RecentEraBabbage -> ledgerBody
         { Babbage.outputs = Babbage.outputs ledgerBody
             <> StrictSeq.fromList
-                ( Ledger.mkSized
-                . Cardano.toShelleyTxOut era
-                . Cardano.toCtxUTxOTxOut
-                . toCardanoTxOut era <$> extraOutputs
-                )
+                (Ledger.mkSized . toBabbageTxOut <$> extraOutputs)
         , Babbage.inputs = Babbage.inputs ledgerBody
             <> Set.fromList (Cardano.toShelleyTxIn <$> extraInputs')
         , Babbage.collateral = Babbage.collateral ledgerBody
             <> Set.fromList (Cardano.toShelleyTxIn <$> extraCollateral')
         , Babbage.txfee =
             modifyFee $ Babbage.txfee ledgerBody
         }
-    ShelleyBasedEraAlonzo -> ledgerBody
+    RecentEraAlonzo -> ledgerBody
         { Alonzo.outputs = Alonzo.outputs ledgerBody
             <> StrictSeq.fromList
-                ( Cardano.toShelleyTxOut era
-                . Cardano.toCtxUTxOTxOut
-                . toCardanoTxOut era <$> extraOutputs
```

Comment:
Avoided going through `cardano-api` types here, and went for `Compatibility.Ledger.toAlonzoTxOut` directly.

Reply:
Btw, some day, we should be able to remove lots of things in the wallet's `Shelley.Compatibility` module.